widgets/bat_openbsd.lua

```diff
@@ -0,0 +1,74 @@
+-- {{{ Grab environment
+local setmetatable = setmetatable
+local tonumber = tonumber
+local string = { format = string.format, gmatch = string.gmatch }
+local io = { popen = io.popen }
+local math = { floor = math.floor }
+local helpers = require("vicious.helpers")
+local string = {
+    gmatch = string.gmatch,
+    match = string.match,
+    format = string.format
+
+}
+-- }}}
+local bat_openbsd = {}
+
+local function worker(format, warg)
+    local battery = warg or "bat0"
+    local filter = string.format("hw.sensors.acpi%s", battery)
+    local cmd = string.format("sysctl -a | grep '^%s'", filter)
+    local proc = io.popen(cmd)
+
+    local bat_info = {}
+    for line in proc:lines("*l") do
+        for key, value in string.gmatch(line, "(%S+)=(%S+)") do
+	    key = key:gsub(filter .. ".", "")
+	    bat_info[key] = value
+        end
+    end
+    proc:close()
+
+    -- current state
+    local states = {[0] = "↯", -- high
+                    [1] = "-", -- low
+                    [2] = "!", -- critical
+                    [3] = "+", -- charging
+                    [4] = "n/a", -- critical
+                    [255] = "⌁"} -- unknown
+    local state = states[tonumber(bat_info.raw0)]
+
+    -- battery capacity in percent
+    local percent = tonumber(bat_info.watthour3 / bat_info.watthour0 * 100)
+
+    local time
+    if tonumber(bat_info.power0) < 1 then
+        time = "∞"
+    else
+	local raw_time = bat_info.watthour3 / bat_info.power0
+	local hours = math.floor(raw_time)
+	local minutes = raw_time % 1
+	time = string.format("%d:%0.2d", hours, minutes)
+    end
+
+    -- calculate wear level from (last full / design) capacity
+    local wear = "N/A"
+    if bat_info.watthour0 and bat_info.watthour4 then
+        local l_full = tonumber(bat_info.watthour0)
+        local design = tonumber(bat_info.watthour4)
+        wear = math.floor(l_full / design * 100)
+    end
+
+    -- dis-/charging rate as presented by battery
+    local rate = bat_info.power0
+
+    -- returns
+    --  * state (high "↯", discharging "-", charging "+", N/A "⌁" }
+    --  * remaining_capacity (percent)
+    --  * remaining_time, by battery
+    --  * wear level (percent)
+    --  * present_rate (W)
+    return {state, percent, time, wear, rate}
```

Comment:
This should be documented in `README.md`.

Reply:
I just added it to the README. The returned table is the same as in the Linux and FreeBSD counterparts (I actually use the same config for both my OpenBSD and my Linux rigs).

Reply:
> I actually use the same config for both my OpenBSD and my Linux rigs

Somehow this makes me extra happy today!